### PR TITLE
Update to v2.10.0

### PR DESCRIPTION
**Pull request by bouweandela**
Update to v2.10.0

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] ~Bumped the build number (if the version is unchanged)~
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by bouweandela** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -59,12 +59,14 @@ outputs:
...
         - nc-time-axis
         - netCDF4
+        - numba
         - numpy !=1.24.3  # severe masking bug
         - openpyxl
         - packaging
         - pandas
         - progressbar2
         - prov
+        - psyplot
```
Comment (schlunma) [inline recipe/meta.yaml:69]: This also appears in l.73, please remove one of them

File: recipe/meta.yaml [hunk 2]
```diff
@@ -59,12 +59,14 @@ outputs:
...
         - psy-maps
         - psy-reg
         - psy-simple
```
Comment (schlunma) [inline recipe/meta.yaml:72]: In environment.yml we have `shapely >=2` (I cannot comment on that line directly).

**Review by schlunma — COMMENTED** [review]
Thanks Bouwe! Two comments.

File: recipe/meta.yaml (hunk 2 above)
Comment (bouweandela) [reply recipe/meta.yaml:72]: Not yet, adding it in https://github.com/ESMValGroup/ESMValTool/pull/3487. But we anyway have the pin via ESMValCore

Comment (schlunma) [reply recipe/meta.yaml:72]: True, I compared against https://github.com/ESMValGroup/ESMValTool/pull/3487.

File: recipe/meta.yaml (hunk 1 above)
Comment (bouweandela) [reply recipe/meta.yaml]: Fixed in [5a4328f](https://github.com/conda-forge/esmvaltool-suite-feedstock/pull/24/commits/5a4328fc706299b9c0321a34ed324c49578da8b3)

**Review by valeriupredoi — APPROVED** [review]
looks good, bud! Possibly force to pin python < 3.12 even if that comes from esmvalcore - mamba may be faster to solve? Not sure. Cheers and Happy Xmas 🍺 🎄

**Review by schlunma — APPROVED** [review]
🚀

**Pull request merged**